controllers/cluster_controller.go

```diff
@@ -108,76 +109,108 @@ func (r *ClusterReconciler) reconcile(ctx context.Context, cluster *capi.Cluster
 		propagationPolicy := metav1.DeletePropagationBackground
 
 		if !r.DryRun {
-			// handle new CAPI clusters created via an App resource (e.g. cluster-aws)
-			if hasChartAnnotations(cluster) {
-
-				app := &gsapplication.App{}
-				err := r.Client.Get(ctx, getChartNamespacedName(cluster), app)
-				if err != nil && !apierrors.IsNotFound(err) {
-					return ctrl.Result{}, err
-				} else if err == nil {
-					selector := labels.NewSelector()
-					clusterNameReq, _ := labels.NewRequirement(label.Cluster, selection.In, []string{cluster.Name})
-					managedByReq, _ := labels.NewRequirement("app.kubernetes.io/managed-by", selection.NotEquals, []string{"Helm"})
-					selector = selector.Add(*clusterNameReq, *managedByReq)
-
-					// Delete cluster Apps
-					{
-						err := r.Client.DeleteAllOf(ctx, &gsapplication.App{}, &client.DeleteAllOfOptions{
-							ListOptions: client.ListOptions{
-								Namespace:     cluster.GetNamespace(),
-								LabelSelector: selector,
-							},
-							DeleteOptions: client.DeleteOptions{
-								PropagationPolicy: &propagationPolicy,
-							},
-						})
-						if err != nil {
-							log.Error(err, fmt.Sprintf("unable to delete Apps for cluster %s/%s", cluster.Namespace, cluster.Name))
-							ErrorsTotal.WithLabelValues(cluster.Name, cluster.Namespace).Inc()
-							return requeue(), nil
-						}
+			if r.CApiProvider == "" {
+				if !hasChartAnnotations(cluster) {
+					err := r.Client.Delete(ctx, cluster, client.PropagationPolicy(propagationPolicy))
+					if err != nil {
+						log.Error(err, fmt.Sprintf("unable to delete cluster %s/%s", cluster.Namespace, cluster.Name))
+						ErrorsTotal.WithLabelValues(cluster.Name, cluster.Namespace).Inc()
+						return requeue(), nil
 					}
+					log.Info(fmt.Sprintf("Cluster %s/%s was deleted", cluster.Namespace, cluster.Name))
+					SuccessTotal.WithLabelValues(cluster.Name, cluster.Namespace).Inc()
+				} else {
+					log.Info("CAPI cluster detected, ignoring")
+				}
+			} else {
+				if hasChartAnnotations(cluster) {
+					log.Info(fmt.Sprintf("Provider = %s", r.CApiProvider))
+
+					app := &gsapplication.App{}
+					err := r.Client.Get(ctx, getClusterAppNamespacedName(cluster), app)
+					if err != nil && !apierrors.IsNotFound(err) {
+						return ctrl.Result{}, err
+					} else if err == nil {
+						if _, ok := app.Labels[fluxLabel]; ok {
```

Comment:
Please comment why this is here (don't delete GitOps-managed resources, ensure MC itself doesn't commit suicide)